controllers/release/adapter_test.go

```diff
@@ -935,6 +872,170 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		})
 	})
 
+	Context("When registerDeploymentData is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no binding", func() {
+			Expect(adapter.registerDeploymentData(nil, nil)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(BeEmpty())
+		})
+
+		It("does nothing if there is no ReleasePlanAdmission", func() {
+			Expect(adapter.registerDeploymentData(snapshotEnvironmentBinding, nil)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(BeEmpty())
+		})
+
+		It("registers the Release deployment data", func() {
+			Expect(adapter.registerDeploymentData(snapshotEnvironmentBinding, releasePlanAdmission)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.Environment).To(Equal(fmt.Sprintf("%s%c%s",
+				releasePlanAdmission.Namespace, types.Separator, releasePlanAdmission.Spec.Environment)))
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(Equal(fmt.Sprintf("%s%c%s",
+				snapshotEnvironmentBinding.Namespace, types.Separator, snapshotEnvironmentBinding.Name)))
+			Expect(adapter.release.IsDeploying()).To(BeTrue())
+		})
+
+		It("should not register the environment if the ReleasePlanAdmission does not reference any", func() {
+			newReleasePlanAdmission := releasePlanAdmission.DeepCopy()
+			newReleasePlanAdmission.Spec.Environment = ""
+
+			Expect(adapter.registerDeploymentData(snapshotEnvironmentBinding, newReleasePlanAdmission)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.Environment).To(Equal(""))
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(Equal(fmt.Sprintf("%s%c%s",
+				snapshotEnvironmentBinding.Namespace, types.Separator, snapshotEnvironmentBinding.Name)))
+			Expect(adapter.release.IsDeploying()).To(BeTrue())
+		})
+	})
+
+	Context("When registerDeploymentStatus is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no binding", func() {
+			Expect(adapter.registerDeploymentStatus(nil)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("does nothing if the binding doesn't have the expected condition", func() {
+			Expect(adapter.registerDeploymentStatus(snapshotEnvironmentBinding)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("registers the deployment status when the binding has been deployed", func() {
```

Comment:
could probably add a test for the 
```
	} else {
		if condition.Reason == applicationapiv1alpha1.ComponentDeploymentConditionErrorOccurred {
			a.release.MarkDeploymentFailed(condition.Message)
			a.release.MarkReleaseFailed("Release deployment failed")
		} else {
			a.release.MarkDeploying(condition.Message)
		}
	}
```
piece of the code

Reply:
Done.